// Copyright (C) 2017-2020 Cohesity, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program. If not, see <https://www.gnu.org/licenses/>.
//
// Author: Zheng Cai (zheng@cohesity.com)
//
// This file defines the common error codes.

package cohesity.apache_util;

message ErrorProto {
  enum Type {
    kNoError = 0;
    kTransportError = 1;
    kTimeout = 2;
    kRetry = 3;
    kCancelled = 4;
    kNonExistent = 5;
    kNotDirectory = 6;
    kNameTooLong = 7;
    kPermissionDenied = 8;
    kInvalid = 9;
    kNotSupported = 10;
    kBufferTooSmall = 11;
    kSizeExceededSystemLimit = 12;
    kAttrNonExistent = 13;
    kAlreadyExists = 14;
    kCannotDelete = 15;
    kPathNotCovered = 16;
    // The following is only used to maintain a count of the number of errors.
    kErrorCount = 17;
  }

  // Error.
  optional Type type = 1 [default = kNoError];

  // A string describing the errors encountered.
  optional string error_detail = 2;
}
